### PR TITLE
ci: add github action workflow to build & push docker image to quay.io

### PR DESCRIPTION
**Pull request by mhofstetter**
This commit adds two GitHub Actions workflows `Docker Dev` & `Docker Release` which builds & pushes the docker image to quay.io.

### CODE REVIEW

File: .github/workflows/docker-dev.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          password: ${{ secrets.QUAY_DEV_TOKEN }}
+
+      - name: Build & Push Docker Image
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825 # v4.1.1
+        with:
+          context: .
+          push: true
+          platforms: linux/amd64
+          tags: |
+            quay.io/${{ github.repository_owner }}/team-manager-dev:${{ steps.tag.outputs.tag }}
```
Comment (aanm) [inline .github/workflows/docker-dev.yaml:54]: to be consistent with the remaining workflows, this should be `team-manager-ci` instead

File: .github/workflows/docker-release.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - 'v*'
+
+# By specifying the access of one of the scopes, all of those that are not specified are set to 'none'.
+permissions:
+  # To be able to access the repository with `actions/checkout`
+  contents: read
+
+jobs:
+  build:
+    name: Build & Push
```
Comment (aanm) [inline .github/workflows/docker-release.yaml:14]: needs a `environment: release`, which is where we will store the credentials

File: .github/workflows/docker-dev.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          persist-credentials: false
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@ecf95283f03858871ff00b787d79c419715afc34 # v2.7.0
+
+      - name: Login to quay.io
+        uses: docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc # v2.2.0
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_DEV_USERNAME }}
```
Comment (aanm) [inline .github/workflows/docker-dev.yaml:44]: `QUAY_DEV_USERNAME` -> `QUAY_CI_USERNAME`

File: .github/workflows/docker-dev.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          password: ${{ secrets.QUAY_DEV_TOKEN }}
```
Comment (aanm) [inline .github/workflows/docker-dev.yaml:45]: `QUAY_DEV_TOKEN` -> `QUAY_CI_TOKEN`

File: .github/workflows/docker-release.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          persist-credentials: false
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@ecf95283f03858871ff00b787d79c419715afc34 # v2.7.0
+
+      - name: Login to quay.io
+        uses: docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc # v2.2.0
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
```
Comment (aanm) [inline .github/workflows/docker-release.yaml:38]: QUAY_RELEASE_USERNAME

File: .github/workflows/docker-release.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+          password: ${{ secrets.QUAY_TOKEN }}
```
Comment (aanm) [inline .github/workflows/docker-release.yaml:39]: QUAY_RELEASE_TOKEN

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by mhofstetter** [comment]
@aanm Thanks for your review and creation of the secrets!

**Review by aanm — APPROVED** [review]

**Pull request merged**